### ISSUE TITLE
Support ExternalAttributes for ZipArchives

### ISSUE THREAD

**Issue by aventurella** [adamhathcock/sharpcompress#357]
Understanding that SharpCompress is not built on top of `System.IO.Compression` I only point out the .NET Core implementation as a point of reference.

This is currently available in .NET Core 2.x (not .NET Standard 2.x for some reason)
https://github.com/dotnet/corefx/pull/18565
https://github.com/dotnet/corefx/issues/17067

It's already somewhat accounted:
https://github.com/adamhathcock/sharpcompress/blob/master/src/SharpCompress/Writers/Zip/ZipCentralDirectoryEntry.cs#L85-L86

**Comment by adamhathcock** [comment]
Yeah probably just needs fleshing out.  Some copy/paste from DNC's code would work too :)
